### ISSUE TITLE
[Bug] react-static build fails with strange errors. yarn build succeeds

### ISSUE THREAD

**Issue by augustosamame** [react-static/react-static#1552]
a very simple project off the basic template fails to build with react-static build
yarn build works fine

### Environment

Run and copy the result of:

`npx envinfo --system --npmPackages react* --binaries --npmGlobalPackages react* --browsers` 

here:

this command failed with zsh: no matches found: react*

I'm including my package.json here:

{
  "name": "react-static-example-basic",
  "private": true,
  "scripts": {
    "start": "react-static start",
    "stage": "react-static build --staging",
    "build": "react-static build",
    "analyze": "react-static build --analyze",
    "serve": "serve dist -p 3000"
  },
  "dependencies": {
    "@reach/router": "^1.2.1",
    "axios": "^0.19.0",
    "react": "^16.9.0",
    "react-dom": "^16.9.0",
    "react-static": "^7.2.0",
    "react-static-plugin-reach-router": "^7.2.0",
    "react-static-plugin-sitemap": "^7.2.0",
    "react-static-plugin-source-filesystem": "^7.2.0"
  },
  "devDependencies": {
    "babel-eslint": "^10.0.2",
    "eslint": "^6.1.0",
    "eslint-config-react-app": "^5.0.1",
    "eslint-config-react-tools": "^1.1.7",
    "eslint-plugin-flowtype": "^4.2.0",
    "eslint-plugin-import": "^2.18.2",
    "eslint-plugin-jsx-a11y": "^6.2.3",
    "eslint-plugin-react": "^7.14.3",
    "eslint-plugin-react-hooks": "^4.0.2",
    "serve": "^11.1.0"
  }
}

### Steps to Reproduce the problem

**Base your steps off of any freshly installed react-static template!**

1.  create a blank or basic template with latest react-static 7.4.2 version
2.  add a couple simple static routes
3.  react-static build --staging --debug

build fails with strange errors like:

TypeError: Failed exporting HTML for URL talent/3318 (__react_static_root__/src/containers/Star): Cannot convert undefined or null to object

### Expected Behavior

Project should build with no errors using react-static build OR yarn build

### Reproducible Demo

I have uploaded a demo repo here: https://github.com/augustosamame/react-static-demo

**Comment by SleeplessByte** [comment]
Do you, per chance have `react-static` installed as _global_? It's possible `react-static` installed globally is a different version. 

- `react-static build` then is using the global version
- `yarn build` then is using the local version

**Comment by augustosamame** [comment]
Yes. I've followed the README instructions which indicate react-static should be installed globally:

`yarn global add react-static`

It installs version 7.4.2

which I then use to create the project:

`react-static create`

and this for some reason installs react-static 7.2.0 as a dependency in package json.

**Comment by augustosamame** [comment]
I have upgraded my local version of react-static to 7.4.2
Still fails with react-static build and works fine with yarn build

Also tried downgrading my global version to 7.2.0 but got this error:

env: node\r: No such file or directory

which seems to happen if the package has been generated in Windows with non-standard line endings (I'm on Mac)

**Comment by SleeplessByte** [comment]
`"^7.2.0"` is not the same as `7.2.0`. You can check your `yarn.lock` (or `package-lock.json`) to see what it was actually resolved as.

**Comment by SleeplessByte** [comment]
> and this for some reason installs react-static 7.2.0 as a dependency in package json.

It's because the template is outdated. It's something we wanted to fix in the next major version akin to `cra`, which uses a scripts package that is NOT installed globally.

Either way. They _should_ behave the same, but often don't when the versions don't match.

**Comment by augustosamame** [comment]
Thanks. So what would be the fix in my case? Should I just keep using yarn build?
I don't want this to sound as a criticism of a great (and open source) library. But it has been a major hassle to get react-static up and running. I tried everything to get this to work, but kept getting different and cryptic errors, with both basic and empty templates. I kept at it and just stumbled into everything working just fine with yarn build, but I would imagine many first-time users just follow the instructions in the Readme, they see it doesn't work and move on.
I would suggest fixing the initial experience so it works out of the box.
